ticket.go

```diff
@@ -20,81 +22,31 @@ type Ticket struct {
 }
 
 // Create a ticket
-func CreateTicket(client *Client, data *Ticket) {
-	values := map[string]interface{}{
-		"auth":             client.Auth,
-        "api_key":          client.APIKey,
-        "customer_id":      data.CustomerId,
-        "ticket_type_id":   data.TicketTypeId,
-        "serial":           data.Serial,
-        "user_id":          data.UserId,
-        "notes":            data.Notes,
-        "title":            data.Title,
-        "status":           data.Status,
-        "device":           data.Device,
-        "imei":             data.IMEI,
-        "location_id":      data.LocationId,
-    }
-
+func CreateTicket(data *Ticket) map[string]interface{} {
     endpoint := APIBaseUrl + "tickets/create"
-
-    Response(values, endpoint)
+    return Response(data, endpoint)
 }
 
 // Retrieve a list of tickets
-func RetrieveTickets(client *Client) {
-	values := map[string]interface{}{
-		"auth": client.Auth,
-        "api_key": client.APIKey,
-    }
-    
+func RetrieveTickets(data *Ticket) map[string]interface{} {
     endpoint := APIBaseUrl + "tickets"
-
-    Response(values, endpoint)
+    return Response(data, endpoint)
 }
 
 // Retrieve a single ticket
-func RetrieveTicket(client *Client, data *Ticket) {
-    values := map[string]interface{}{
-		"auth": client.Auth,
-        "api_key": client.APIKey,
-    }
-
+func RetrieveTicket(data *Ticket) map[string]interface{} {
     endpoint := APIBaseUrl + "tickets/" + strconv.Itoa(data.Id)
-
-    Response(values, endpoint)
+    return Response(data, endpoint)
 }
 
 // Update a ticket
```

Comment:
A lot of redundant comments, if you're looking to document the code for further explanation, I'd look into Doc generation libraries you can integrate with. That way, if you feel the need to go into detail on certain functions, you can do so without cluttering the code